olive/engine/engine.py

```diff
@@ -558,23 +540,21 @@ def run_no_search(
                     overwrite=True,
                     cache_dir=self._config.cache_dir,
                 )
-                output_model_ids.append(pass_output_model_id)
+                output_models[pass_output_model_id] = output_model_json
 
             # save the evaluation results to output_dir
             if signal is not None:
                 results_path = output_dir_with_pf / f"{final_output_name}_metrics.json"
                 with open(results_path, "w") as f:
                     json.dump(signal.to_json(), f, indent=4)
 
-            if output_model_json and not should_prune:
-                # output_model_json is the last model only if the flow is not pruned
-                output["model"] = output_model_json
-                if signal is not None:
-                    output["metrics"] = signal
-            else:
-                output = None
-            flows_output[tuple(pass_flow)] = output
-        return flows_output, output_model_ids
+        output_model_ids = list(output_models.keys())
+        fp_outputs = deepcopy(self.footprints[accelerator_spec].create_footprints_by_model_ids(output_model_ids))
```

Comment:
>deepcopy [](http://example.com/codeflow?start=21&length=8)

why do we need deepcopy? I suppose the create method will return new object. right?

Reply:
Good catche. The create method is renamed from get_footprints_by_model_ids where the dict is used to assign to a new variable, but not the new object. I will try to move the deepcopy logics to creat methods.